Proj1 multiple asic cfg

J0nPatrik opened the pull request:
Prepared two new commands that should handle storing and applying ASIC configuration in the NVM memory.

J0nPatrik committed:
Added two new commands for ASIC
configuration management

Added two new UART commands
 - CMD_SEND_NVM_CITIROC_CONF:
   Used to store ASIC configuration on the NVM memory
 - CMD_APPLY_NVM_CITIROC_CONF:
   Used to get a configuration from the NVM memory
   and apply it on the ASIC

patrikj5 committed:
Define name change and extended commanddata

Define namechange
 - MSP_OP_APPLY_CUBES_CITI_CONF -> MSP_OP_SELECT_NVM_CITI_CONF

Extendend array commanddata to 144 to fit last byte of new command

J0nPatrik committed:
Updated define
CMD_SELECT_NVM_CONF: 'n' -> 'N'

J0nPatrik committed:
Fixed switch statement bug
Added missing breaks in switch statement in function RS_read

tstana commented on obcsim/RS232.cpp:
```diff
@@ -52,6 +52,24 @@ void RS_read(msp_link_t *lnk)
+      if (fill_commanddata(len)) {
+        Serial.println("-------- Invoking SEND_NVM_CITI_CONF -----");
+        invoke_send(lnk, MSP_OP_SEND_NVM_CITI_CONF, commanddata, len, BYTES);
+        Serial.println("--------------------------------------");
```
Please add a few more dashes here so that the column where this print ends matches the "Invoking" print above. With the current code, what will get printed is:

```
-------- Invoking SEND_NVM_CITI_CONF -----
--------------------------------------
```

whereas what we are looking for (to keep to the conventions already present in the code) is:

```
-------- Invoking SEND_NVM_CITI_CONF -----
------------------------------------------
```

tstana commented on obcsim/RS232.cpp:
```diff
@@ -52,6 +52,24 @@ void RS_read(msp_link_t *lnk)
+      if (fill_commanddata(len)) {
+        Serial.println("-------- Invoking SELECT_NVM_CITIROC_CONF -----");
+        invoke_send(lnk, MSP_OP_SELECT_NVM_CITI_CONF, commanddata, len, BYTES);
+        Serial.println("--------------------------------------");
```
Same comment as above for `SEND_NVM_CITIROC_CONF`.

Tip (maybe obvious, but hey): You can use the column of the print above to align where the dashes should end up.

tstana commented on obcsim/obcsim_configuration.hpp:
```diff
@@ -49,6 +49,12 @@
 #ifndef MSP_OP_SEND_CUBES_GATEWARE_CONF
 #define MSP_OP_SEND_CUBES_GATEWARE_CONF 0x77
 #endif
+#ifndef MSP_OP_SEND_NVM_CITI_CONF
+#define MSP_OP_SEND_NVM_CITI_CONF 0x78
+#endif
+#ifndef MSP_OP_SELECT_NVM_CITI_CONF
+#define MSP_OP_SELECT_NVM_CITI_CONF 0x79
+#endif
```
These will conflict with what has already been added by @IHamed98 as part of his PR, which has already been merged. Please use `0x79` and `0x7A`.

tstana reviewed (changes requested):
@J0nPatrik, this looks good, but there are some minor changes still needed, to keep to the conventions already established and -- more importantly -- avoid a merge conflict later on.

J0nPatrik committed:
Fixed conventions

- Fixed Serial.println statements so that print statements match in lenght
- Changed address for two obcsim_configuratinos to address merge issues

J0nPatrik committed:
Merge branch 'proj1_multiple_asic_cfg' of https://github.com/tstana/proto-cubes-obc into proj1_multiple_asic_cfg

tstana commented on obcsim/obcsim_configuration.hpp:
```diff
@@ -49,6 +49,12 @@
+#define MSP_OP_SEND_NVM_CITI_CONF 0x79
+#endif
+#ifndef MSP_OP_APPLY_CUBES_CITI_CONF
+#define MSP_OP_APPLY_CUBES_CITI_CONF 0x7A
```
Should be `MSP_OP_SELECT_CUBES_CITI_CONF`, to be in concordance to the definition in `cubes-fw`. (Sorry, I guess I missed this last time.)

tstana commented on obcsim/RS232.cpp:
```diff
@@ -52,6 +52,24 @@ void RS_read(msp_link_t *lnk)
+      Serial.println("CMD_SELECT_NVM_CITIROC_CONF received");
+      len = 1;
+      if (fill_commanddata(len)) {
+        Serial.println("-------- Invoking APPLY_NVM_CITIROC_CONF -----");
```
Should be `MSP_OP_SELECT_CUBES_CITI_CONF`, to be in concordance to the definition in cubes-fw. (Sorry, I guess I missed this last time.)

J0nPatrik committed:
Define namechange

MSP_OP_APPLY_CUBES_CITI_CONF -> MSP_OP_SELECT_CITI_CONF
